hnix-store-core/src/System/Nix/StorePathMetadata.hs

```diff
@@ -47,3 +52,42 @@ data StorePathTrust
   | -- | It was built elsewhere (and substituted or similar) and so
     -- is less trusted
     BuiltElsewhere
+
+data NarInfo
+  = NarInfo
+  { -- | URL to compressed Nar file.
+    narUrl :: Text
+  , -- | Form of compression used on Nar file.
+    narCompression :: Text
+  , -- | The hash of the nar serialization of the path.
+    fileHash :: !SomeNamedDigest
+  , -- | The size of the nar the path, in bytes.
+    fileBytes :: !(Maybe Word64)
+  }
+
```

Comment:
I've completely missed this and recently wrote/extracted from Cachix a lib - https://github.com/sorki/nix-narinfo which is hopefully polymorphic enough to accommodate our types.